app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -29,10 +29,10 @@ def customize_destination
     signal :post_create_destination
   end
 
-  def find_destination_in_vmdb(_ems_ref)
+  def find_destination_in_vmdb(ems_ref)
     return if phase_context[:cloud_api_completion_time].nil? || source.ext_management_system.last_refresh_date < phase_context[:cloud_api_completion_time]
 
-    source.ext_management_system&.vms_and_templates&.find_by(:ems_id => options[:src_ems_id].try(:first), :name => options[:vm_name], :template => (request_type == 'clone_to_template'))
+    source.ext_management_system&.vms_and_templates&.find_by(:ems_id => options[:src_ems_id].try(:first), :ems_ref => ems_ref, :template => (request_type == 'clone_to_template'))
```

Comment:
@miyamotoh the `find_destination_in_vmdb` override was added as part of #374. The [base method](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/provision/cloning.rb?plain=1#L3) uses `:ems_ref`, not `:name`. Can you remember if there is a reason we wanted to use `:name` here?

Reply:
No clear memory, but I think I was trying to get `find_by` to work with the info I had, likely got working with _easy_ params, and never went back to find the _best_ way of doing it. I agree `ems_ref` would probably be the best param to use here. 👍

Reply:
Having both `source.ext_management_system.vms_and_templates` _and_ `:ems_id => options[:src_ems_id].try(:first)` is redundant.

The `ext_management_system.vms_and_templates` association already includes the `ems_id` in the query.
```suggestion
    source.ext_management_system&.vms_and_templates&.find_by(:ems_ref => ems_ref, :template => (request_type == 'clone_to_template'))
```

Reply:
Is it possible to have an instance and a template with the same `ems_ref`?  (aka do we need `:template => (request_type == 'clone_to_template')` at all?)

Reply:
> Is it possible to have an instance and a template with the same `ems_ref`?

I _think_ the IDs are globally unique but that is __not__ documented so I think in this case it is safer to leave it in.